mediaire_toolbox/data_cleaner.py

```diff
@@ -75,9 +132,22 @@ def clean_up(self, dry_run=False):
                     "Current total size is too big (%s bytes), need to delete some data to free up space" % current_size)
             if delete:
                 removed.append(folder)
+                pre_clean_size = self.current_size(folder)
                 if dry_run:
                     default_logger.info('(dry-run) Would remove folder [%s]' % folder)
+                    if self.whitelist or self.blacklist:
+                        self.clean_folder(folder)
                 else:
                     default_logger.info('Removing folder [%s]' % folder)
-                    shutil.rmtree(folder)
+                    if self.whitelist or self.blacklist:
+                        self.clean_folder(folder)
```

Comment:
What happens if the whitelist or blacklist is such that all the contents of the folder are deleted? In that case would we keep empty folders? Let's avoid that and also make sure that we remove empty folders